schema/crdb/dbinit.sql

```diff
@@ -4031,6 +4031,56 @@ CREATE TABLE IF NOT EXISTS omicron.public.cockroachdb_zone_id_to_node_id (
     PRIMARY KEY (omicron_zone_id, crdb_node_id)
 );
 
+/*
+ * List of debug datasets available for use (e.g., by support bundles).
+ *
+ * This is a Reconfigurator rendezvous table: it reflects resources that
+ * Reconfigurator has ensured exist. It is always possible that a resource
+ * chosen from this table could be deleted after it's selected, but any
+ * non-deleted row in this table is guaranteed to have been created.
+ */
+CREATE TABLE IF NOT EXISTS omicron.public.rendezvous_debug_dataset (
+    /* ID of the dataset in a blueprint */
+    id UUID PRIMARY KEY,
+
+    /* Time this dataset was added to the table */
+    time_created TIMESTAMPTZ NOT NULL,
+
+    /*
+     * If not NULL, indicates this dataset has been expunged in a blueprint.
+     * Multiple Nexus instances operate concurrently, and it's possible any
+     * given Nexus is operating on an old blueprint. We need to avoid a Nexus
+     * operating on an old blueprint from inserting a dataset that has already
+     * been expunged and removed from this table by a later blueprint, so
+     * instead of hard deleting, we tombstone rows via this column.
+     *
+     * Hard deletion of tombstoned datasets will require some care with respect
+     * to the problem above. For now we keep tombstoned datasets around forever.
+     */
+    time_tombstoned TIMESTAMPTZ,
```

Comment:
It may be worth adding a comment here or in the corresponding rust type that we should not rely on the fact that time_tombstoned is later than time_created for correctness. While it's highly unlikely that is going to not be true, it still shouldn't be relied upon.

Reply:
Hmm. Do you think I should add a `blueprint_when_tombstoned` column? That's what we're really operating on; the time is only useful to a human to narrow down when things might have happened.

Reply:
I think that would indeed be useful. I don't think you want to get rid of the timestamps, just add that aditional column and a comment. Thank you!

Reply:
Added in 8372bb69b